src/Disks/DiskLocal.cpp

```diff
@@ -317,64 +367,160 @@ SyncGuardPtr DiskLocal::getDirectorySyncGuard(const String & path) const
     return std::make_unique<LocalDirectorySyncGuard>(fs::path(disk_path) / path);
 }
 
-DiskPtr DiskLocalReservation::getDisk(size_t i) const
+bool DiskLocal::setupAndCheck()
 {
-    if (i != 0)
+    try
     {
-        throw Exception("Can't use i != 0 with single disk reservation", ErrorCodes::INCORRECT_DISK_INDEX);
+        fs::create_directories(disk_path);
+    }
+    catch (...)
+    {
+        LOG_WARNING(logger, "Cannot create the directory of disk {} ({}).", name, disk_path);
+        return false;
```

Comment:
I think that it should fail if setupAndCheck failed for the first time, consider the case when the permissions for the directory is incorrect, ignoring such disks does not sounds correct behavior to me, the server should fail I think (otherwise it will hide this error and user may realize that not all disks are used only a year)

Reply:
But it's a common case that broken disks aren't recovered yet and we restart.